src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -21,19 +27,38 @@ function useChatUtilsHook(): UseChatUtilsHook {
   const {localDb, refresh} = useLocalDatabaseHook();
   const navigation = useNavigation();
   const {selectedChannel} = chat;
-
-  const setChannelAsRead = (channel: ChannelList) => {
+  const [profile] = React.useContext(Context).profile;
+  const setChannelAsRead = async (channel: ChannelList) => {
     if (!localDb) return;
     channel.setRead(localDb).catch((e) => console.log('setChannelAsRead error', e));
 
-    AnonymousMessageRepo.setChannelAsRead(channel?.id).catch((e) => {
-      console.log('setChannelAsRead error api', e?.response?.data);
-    });
+    if (channel?.channelType?.includes('ANON')) {
+      try {
+        await AnonymousMessageRepo.setChannelAsRead(channel?.id?.replace('_anon', ''));
+      } catch (error) {
+        console.log('setAnonChannelAsRead error api', error);
+      }
+    } else {
+      const channelType = {
+        messaging: ChannelTypeEnum.Messaging,
+        group: ChannelTypeEnum.Group,
+        topics: ChannelTypeEnum.Community
+      };
+
+      try {
+        await SignedMessageRepo.setChannelAsRead(
+          channel?.id,
+          channelType[channel?.rawJson?.channel?.type]
+        );
+      } catch (error) {
+        console.log('setSignedChannelAsRead error api', error);
+      }
+    }
 
     refresh('channelList');
   };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling in this section could be improved. Currently, errors are just logged to the console and not handled further. This might lead to unexpected behavior in your application if an error occurs. Consider adding more robust error handling here.
```diff
     channel.setRead(localDb).catch((e) => console.log('setChannelAsRead error', e));
...
        console.log('setAnonChannelAsRead error api', error);
...
        console.log('setSignedChannelAsRead error api', error);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->